runtime/lua/vim/lsp/handlers.lua

```diff
@@ -53,7 +53,7 @@ local function progress_handler(_, result, ctx, _)
       end
     end
   else
-    table.insert(client.messages, {content = val, show_once = true, shown = 0})
+    table.insert(client.messages.messages, {content = val, show_once = true, shown = 0})
```

Comment:
Could this instead add a `.progress` entry that has `done = true` ? The `.messages.messages` repetition feels a bit odd.

Reply:
You mean I should shove those messages into `client.messages.progress` instead and handle them like progress messages too?

Reply:
Yes, if that works

Reply:
So I just looked over the spec again and I'm not sure how it could even happen to receive non-progress messages via the progress handler. That matches what I've seen with sumneko, rust-analyzer so far—the `messages.messages` table has always been empty.

Do you know of any language servers which send non-progress messages via `$/progress`?

Reply:
I spoke to @wbthomason and this is what they had to say:

> I've only found those non-standard extensions in pyls and clangd - as far as I know, everything else just uses $/progress messages from https://microsoft.github.io/language-server-protocol/specification#progress  
> That said, I never went looking for other standard-extending servers once I got those two working, since all the others I needed to use (e.g. rust-analyzer, JuliaLS, sumneko, etc.) used the standard if they reported progress at all  
> I think it's probably safe to remove the non-progress message parts and leave those extensions to non-core plugins, e.g. lsp-status or clangd_extensions

So I'll just remove them.

Reply:
The spec doesn't list any kind of payload structure at all, so whether they contain `kind` or not is undefined. But they should always contain a `token`:

```
type ProgressToken = integer | string;

interface ProgressParams<T> {
	/**
	 * The progress token provided by the client or server.
	 */
	token: ProgressToken;

	/**
	 * The progress data.
	 */
	value: T;
}
```

This is why I would add messages without kind to progress under the token with `done = true`

Reply:
Ahh, now I understand. Will do!

Reply:
Alright, done.